common/persistence/nosql/nosqlplugin/cassandra/translator/fixed_address_translator.go

```diff
@@ -0,0 +1,84 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package translator
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/gocql/gocql"
+	"go.temporal.io/server/common/config"
+)
+
+const (
+	fixedTranslatorName   = "fixed-address-translator"
+	advertisedHostnameKey = "advertised-hostname"
+)
+
+func init() {
+	RegisterTranslator(fixedTranslatorName, NewFixedAddressTranslatorPlugin())
+}
+
+type FixedAddressTranslatorPlugin struct {
+}
+
+func NewFixedAddressTranslatorPlugin() TranslatorPlugin {
+	return &FixedAddressTranslatorPlugin{}
+}
+
+func (plugin *FixedAddressTranslatorPlugin) GetTranslator(cfg *config.Cassandra) (gocql.AddressTranslator, error) {
+	if cfg.AddressTranslator == nil {
+		return nil, errors.New("there is no addressTranslator configuration in cassandra configuration")
+	}
+
+	opts := cfg.AddressTranslator.Options
+	if opts == nil {
+		return nil, errors.New("there are no options for translator plugin")
+	}
+
+	advertisedHostname, found := opts["advertised-hostname"]
+
+	if !found || len(advertisedHostname) == 0 {
+		return nil, errors.New("detected no advertised-hostname key or empty value for translator plugin options")
+	}
+
+	var resolvedIp net.IP = nil
+	ips, _ := net.LookupIP(advertisedHostname)
+	for _, ip := range ips {
+		if ipv4 := ip.To4(); ipv4 != nil {
+			resolvedIp = ipv4
+			break
+		}
+	}
+
+	if resolvedIp == nil {
+		return nil, fmt.Errorf("no resolved IP for advertised hostname %q", advertisedHostname)
+	}
+
+	return gocql.AddressTranslatorFunc(func(addr net.IP, port int) (net.IP, int) {
+		return resolvedIp, port
```

Comment:
instead of using the fixed one, why not doing the translate here? I imaging that if the underline IP changed, the gocql would call this translator again, and we should do the lookup here again as well.